### DISCUSSION TITLE
route mavlink trafic from ethernet to wifi

### DISCUSSION THREAD

**Discussion by ozitoune** [mavlink-router/mavlink-router#410] [Q&A]
Hello i have a pixhawk 6x connected to my raspbery pi 4 via ethernet. I recive mavlink on my raspbery  on the port 14550
How to route mavlink traffic from the thernet to the wifi connection?  (the raspberry is connected to a wifi routeur)  I need to connect 2 computers (on the same wifi network than the raspbery is connected) with qgroundcontrol to my pixhawk, is it possible ?
Thanks for your help !

**Comment by jbeyerstedt** [comment]
With one QGC instance you could just let mavlink-router forward UDP data to its magic MAVLink port. But I wouldn't recommend it and you need to connect two QGC instances anyways.

So I would recommend to use the TCP server functionality of mavlink-router and let QGC establish a TCP connection. You can do this in "Application Settings" > "Comm Links" and then TCP to the IP address of your raspberry Pi.
mavlink-router should have it's TCP server enabled by default and listening on port 5760. So if you have mavlink-router running, this shall work out-of-the-box.
But it's important, that all QGC instances have different MAVLink System IDs! This can be changed in the settings as at the "MAVLink" panel/section.

**Reply by ozitoune** [reply]
thanks for reply ! 
I have my ethernet raspbery adress to 192.168.0.101 connected by ethernet to my pixhawk 6x (192.168.0.5). The raspbery is connected to my wifi routeur on 192.168.1.101, and my computer with qground control is 192.168.1.102. (on the same wifi network)
so if i try tcp connexion on qgroundcontrol to server 192.168.1.101 and tcp 5760, nothings happen, and qground control no connect to my pixhawk :( 
what is the command to do that ?

**Reply by jbeyerstedt** [reply]
Is mavlink-router running on the raspberry pi and how is it configured?

**Comment by ozitoune** [comment]
yes but i don't know wich command i have to enter, i have tried mavlink-routerd 0.0.0.0:14550, mavlink-routerd 192.168.1.101:14550,  mavlink-routerd 0.0.0.0:14550 -e 192.192.168.1.102, mavlink-routerd 192.168.0.101:14550,...  i not understand where to specifie the input, and the output, in my case the raspberry recive mavlink from ethernet by pixhawk 6x on 192.168.0.101, and i want my raspbery transmit mavlink by wifi, the adress off the raspbery wifi , is 192.168.1.101,  and i want to connect my computer by wifi, the adress is 192.168.1.102...

**Reply by jbeyerstedt** [reply]
> I don't understand where to specify the input and the output

It's all in here: https://github.com/mavlink-router/mavlink-router#endpoints. With UDP connections, the "client" must always be the system which is sending messages on its own. PX4 will just start sending heartbeats, but mavlink-router and QGC won't.

So your pixhawk needs to be setup to send MAVLink messages to a UDP "server" mode endpoint on your raspberry pi and mavlink-router can listen on this port like this: `mavlink-routerd 0.0.0.0:14550`. (With `-e` you can specify where mavlink-router will send data to, but we use the TCP server for that because it's more reliable with multiple QGC instances.)

**Reply by jbeyerstedt** [reply]
I would recommend to start mavlink-router with the `-v` flag to increase verbosity. Then you can take a look at the incoming packets and where they are routed to. As described in https://github.com/mavlink-router/mavlink-router#message-routing, normal telemetry will be "broadcast" on all endpoints while targeted messages (like parameter fetching) need to know which system is connected to which endpoint. The verbose mode will also show you the list of connected systems.

**Comment by ozitoune** [comment]
sometimes i see TCP [9] dynamic: Connexion accepted , when i put tcp connexion and 192.168.1.101 , but in qground control no connexion to vehicle, and sometimes i see : 5 messages to unknow endpoints in the last 5 seconds...when i try udp connexion to 192.168.1.101:14550

**Comment by ozitoune** [comment]
ok so i launch mavlink-routeurd 0.0.0.0:14550 on the raspbery ,an  i my computer i put 192.168.1.101 on tcp connexion 5760, is it correct ?? Because i see connection accepted on the raspbery terminal where i launch mavlink-routerd 0.0.0.0:14550 but qgroundcontrol never connect the vehicle :(  nothing happens

**Comment by ozitoune** [comment]
Is working now with tcp after rebooting my raspbery , i don't know why but after reboot all is ok  ! sory for your time, and thanks a lot ....
Another question, i have another pixhawk without ethernet , so if i connect it by usb the command to do the same will be mavlink-routerd 0.0.0.0:14550 /dev/ttyUSB0:57600  or  mavlink-routerd 192.168.1.101:14550  /dev/ttyUSB0:57600  ?
Thank you

i
